src/score_hv/harvesters/daily_bfg.py

```diff
@@ -141,107 +231,154 @@ def get_data(self):
-                longname = "Top of atmosphere net radiative energy flux"
-                units = "W/m**2"
-                
-                try:
-                    temporal_means = (
-                        np.ma.masked_invalid(xr_dataset['dswrf_avetoa'].mean(
-                                                                dim='time',
-                                                                skipna=True)) - 
-                        np.ma.masked_invalid(xr_dataset['uswrf_avetoa'].mean(
-                                                                dim='time',
-                                                                skipna=True)) - 
-                        np.ma.masked_invalid(xr_dataset['ulwrf_avetoa'].mean(
-                                                                dim='time',
-                                                                skipna=True)))
-                except KeyError as err:
-                    msg = (f'{xr_dataset.data_vars} '
-                           'do not include all '
-                           'variables needed to calculate TOA net '
-                           'radiative energy flux '
-                           '("dswrf_avetoa," "uswrf_avetoa" and '
-                           '"ulwrf_avetoa")')
-                    raise KeyError(msg) from err
-                                                                
-            else:
-                variable_data = xr_dataset[variable]
-                if 'long_name' in variable_data.attrs:
-                    longname = variable_data.attrs['long_name']
+              The temporal means is a list which will hold the 
+              temporal means calculated in this function.
+              The temporal means are passed into the stats_util 
+              class in order to calculate the statistics requested
+              by the user.
+              """
+            temporal_means = []
+            region_values = regions_instance.get_user_region()
+            for region in region_values:
+                name = region['name']
+                min_lat = region['min_lat']
+                max_lat = region['max_lat']
+                east_lon = region['east_lon']
+                west_lon = region['west_lon']
+                desired_latitude_indices=[index for index, lat in enumerate(latitudes) if min_lat <= lat <= max_lat]
+                if desired_latitude_indices:
+                   lat_start_index=desired_latitude_indices[0]
+                   lat_end_index=desired_latitude_indices[-1]
                 else:
-                    longname = "None"
-                if 'units' in variable_data.attrs:
-                    units = variable_data.attrs['units']
+                   msg=f"No latitude values found within the specified range of {min_lat} and {max_lat}."
+                   raise KeyError(msg)
+
+                desired_longitude_indices=[index for index, lon in enumerate(longitudes) if east_lon <= lon <= west_lon]
```

Comment:
I think there may be issues with this code in the future because we don't guarantee that east_lon is less than west_lon and we don't want to so that we can have regions surrounding the prime meridian.  can we update this to handle these cases?